pkg/util/wait/backoff.go

```diff
@@ -0,0 +1,73 @@
+package wait
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/utils/clock"
+	"k8s.io/utils/ptr"
+)
+
+// UntilWithBackoff runs f in a loop until context indicates finished. It
+// applies backoff depending on the SpeedSignal f returns.  Backoff increases
+// exponentially, ranging from 1ms to 100ms.
+func UntilWithBackoff(ctx context.Context, f func(context.Context) SpeedSignal) {
```

Comment:
```suggestion
func UntilWithBackoff(ctx context.Context, f func(context.Context) bool) {
```

The function can simply return whether to backoff or not. Or from the scheduler's perspective: whether it successfully scheduled anything.

Reply:
Is a bool better than a specific type? I defined this type so that the API would be self-explanatory (and harder to mistakenly flip the return type)

Reply:
We can use bools when there is no ambiguity of what the return value is. But in this case, it is a bit ambiguos.

Reply:
@gabesaba Do you have more advantages rather than using `bool`?

Reply:
For type safety: to make UntilWithBackoff harder to use incorrectly. When users read code, it will be clear that this return value means to backoff or not, rather than having to look up in library definition of UntilWithBackoff when true/false should be returned.

Especially since there's some indirection in the usage: `Scheduler.schedule` implements the API, and this function is provided to `UntilWithBackoff` by `Scheduler.Start`